packages/frontend/core/src/blocksuite/presets/ai/chat-panel/components/doc-chip.ts

```diff
@@ -103,15 +109,22 @@ export class ChatPanelDocChip extends SignalWatcher(
       if (!doc.ready) {
         doc.load();
       }
-      const result = await extractMarkdownFromDoc(doc, this.host.std.provider);
-      if (this.chip.markdown) {
-        this.chip.markdown.value = result.markdown;
+      const value = await extractMarkdownFromDoc(doc, this.host.std.provider);
+      const tokenCount = estimateTokenCount(value);
+      if (this.checkTokenLimit(this.chip, tokenCount)) {
+        const markdown = this.chip.markdown ?? new Signal<string>('');
+        markdown.value = value;
+        this.updateChip(this.chip, {
+          state: 'success',
+          markdown,
+          tokenCount,
+        });
       } else {
-        this.chip.markdown = new Signal<string>(result.markdown);
+        this.updateChip(this.chip, {
+          state: 'failed',
+          tooltip: 'Content touched the token limit',
```

Comment:
rephrase to `Doc content exceeds the maximum token limit` instead

Reply:
This is the copy on the design draft. [Take a look](https://www.figma.com/design/sYt0WsbCjeLlougFcTLO5Z/%F0%9F%9B%BC-2024-S7---0.20?node-id=2557-199784&t=WE2Yjrh1S0ii4lnW-0).

Reply:
IMHO `touched` should be replaced by `exceed`, `reached` or `hit`.

Reply:
You are right. Thank you, xp! Please take another look.